### PR TITLE
[SDK-3703] Feature/integration 2.0

### LINKED ISSUE #1244: Review and merge integration 2.0
Related to #1199
- Sync with the main.
- Need to re-review and add missing tests ( if needed ).
- Merge with main.

### PR DESCRIPTION
**Pull request by sacOO7**
- Implemented no connection serial as per https://github.com/ably/specification/pull/88/files 
- Deprecated the recoveryKey support as per https://github.com/ably/ably-dotnet/issues/1187
- For more details -> https://github.com/ably/ably-js/pull/1110
- Fixed #1244

### CODE REVIEW

**Review by SimonWoolf — APPROVED** [review]
Not very familar with .net, but afaict this looks good to me

File: src/CommonAssemblyInfo.cs [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 [assembly: AssemblyCompany("Ably")]
 [assembly: AssemblyProduct("Ably .NET Library")]
-[assembly: AssemblyVersion("1.2.9")]
-[assembly: AssemblyFileVersion("1.2.9")]
+[assembly: AssemblyVersion("2.0.0")]
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply src/CommonAssemblyInfo.cs:7]: Seems we don't need to change it. I will update according to minor version change 👍

[the remark this answers is not in the document]
Comment (sacOO7) [reply src/CommonAssemblyInfo.cs]: I have removed the changes from the file.

File: src/IO.Ably.Shared/Realtime/ChannelMessageProcessor.cs [hunk 2]
```diff
@@ -63,7 +74,7 @@ public Task<bool> MessageReceived(ProtocolMessage protocolMessage, RealtimeState
                         // RTL12
                         if (!protocolMessage.HasFlag(ProtocolMessage.Flag.Resumed))
                         {
-                            channel.Presence.ChannelAttached(protocolMessage);
+                            channel.Presence.HandleAlreadyAttachedChannel(protocolMessage);
```
Comment (sacOO7) [inline src/IO.Ably.Shared/Realtime/ChannelMessageProcessor.cs:77]: Need to check if the presence needs to start sync only when the channel was already attached. 
As per `ably-go` implementation, this is more simplified. 

This has nothing to do with `spec-2.0`, rather might be a existing bug in the library,

Comment (sacOO7) [reply src/IO.Ably.Shared/Realtime/ChannelMessageProcessor.cs]: This has been re-implemented : )

File: src/IO.Ably.Shared/Realtime/PresenceMap.cs [hunk 3]
```diff
@@ -121,15 +138,16 @@ public void StartSync()
...
-            if (!IsSyncInProgress)
+            if (!SyncInProgress)
             {
                 lock (_lock)
                 {
-                    _residualMembers = new HashSet<string>(_members.Keys);
-                    IsSyncInProgress = true;
+                    _beforeSyncMembers = new HashSet<string>(_members.Keys); // RTP19
+                    SyncInProgress = true;
+                    SyncCompleted = false;
```
Comment (SimonWoolf) [inline src/IO.Ably.Shared/Realtime/PresenceMap.cs:150]: Why do we need both of these? they're always the opposite of each other, right?

**Review by SimonWoolf — APPROVED** [review]
Looks good to me, one small comment

File: src/IO.Ably.Shared/Realtime/PresenceMap.cs (hunk 3 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/Realtime/PresenceMap.cs:150]: so, I have to maintain an enum stating 3 states. initial, progress and completed

**Pull request merged**